version: "3.9"

services:
  postgres_db:
    image: postgres:14.8-alpine
    restart: always
    volumes:
      - bot_db:${DB_ROOT_PATH}
      - ./backup:${DB_ROOT_PATH}/backup
    ports:
      - "5432:5432"
    env_file:
      - .env
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U $POSTGRES_USER -d $POSTGRES_DB"]
      interval: 5s
      retries: 5
      timeout: 3s

  migration:
    image: migrate/migrate
    restart: on-failure
    command: --path=/migrations/ --database="${DB_CONNECTION}" up
    volumes:
      - ./migrations:/migrations
    env_file:
      - .env
    depends_on:
      postgres_db:
        condition: service_healthy

  bot:
    restart: "no"
    build:
      context: .
    env_file:
      - .env
    volumes:
      - ./logs:${ROOT_PATH}/logs
      - ./backup:${ROOT_PATH}/backup
    depends_on:
      postgres_db:
        condition: service_healthy
      migration:
        condition: service_completed_successfully

volumes:
  bot_db:
    external: true
